libcaf_io/src/tcp.cpp

```diff
@@ -43,11 +43,12 @@ rw_state tcp::read_some(size_t& result, native_socket fd, void* buf,
   CAF_LOG_TRACE(CAF_ARG(fd) << CAF_ARG(len));
   auto sres = ::recv(fd, reinterpret_cast<io::network::socket_recv_ptr>(buf),
                      len, no_sigpipe_io_flag);
-  CAF_LOG_DEBUG(CAF_ARG(len) << CAF_ARG(fd) << CAF_ARG(sres));
   if (is_error(sres, true) || sres == 0) {
     // recv returns 0  when the peer has performed an orderly shutdown
+	CAF_LOG_DEBUG(CAF_ARG(len) << CAF_ARG(fd) << CAF_ARG(sres));
     return rw_state::failure;
   }
+  CAF_LOG_DEBUG(CAF_ARG(len) << CAF_ARG(fd) << CAF_ARG(sres));
```

Comment:
> Alternatively, we could restrict logging to the failure case and use CAF_LOG_ERROR instead?

I agree. If we move the logging further down and it makes sense to log errors and ordinary program flow separately. The previous debug log entries are irrelevant in case of an error and we can instead output the socket error directly.